certora/specs/Roles.spec

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+methods {
+    function owner() external returns address envfree;
+    function curator() external returns address envfree;
+    function guardian() external returns address envfree;
+    function isAllocator(address target) external returns bool envfree;
+
+    function _.idToMarketParams(MetaMorphoHarness.Id) external => AUTO;
+    function _.supplyShares(MetaMorphoHarness.Id, address) external => AUTO;
+    function _.accrueInterest(MetaMorphoHarness.MarketParams) external => AUTO;
+
+    function _.balanceOf(address) external => AUTO;
+
+    function SafeERC20.safeTransfer(address, address, uint256) internal => CONSTANT;
+    function SafeERC20.safeTransferFrom(address, address, address, uint256) internal => CONSTANT;
+}
+
+rule curatorIsAllocator(method f, calldataarg args)
+filtered {
+    f -> !f.isView
+    && f.selector != sig:updateWithdrawQueue(uint256[]).selector
+    && f.selector != sig:deposit(uint256, address).selector
+    && f.selector != sig:redeem(uint256, address, address).selector
+    && f.selector != sig:transfer(address, uint256).selector
+    && f.selector != sig:skim(address).selector
+    && f.selector != sig:acceptCap(MetaMorphoHarness.MarketParams).selector
+    && f.selector != sig:transferFrom(address, address, uint256).selector
+    && f.selector != sig:reallocate(MetaMorphoHarness.MarketAllocation[]).selector
+    && f.selector != sig:mint(uint256, address).selector
+    && f.selector != sig:withdraw(uint256, address, address).selector
+    && f.selector != sig:multicall(bytes[]).selector
+    && f.selector != sig:submitCap(MetaMorphoHarness.MarketParams, uint256).selector
+    && f.selector != sig:acceptOwnership().selector
```

Comment:
why do we have to skip this actually ?

Reply:
There is a pending discussion with the Certora team about how to handle summarized calls: they seem to always revert when their returned values is used, and I don't know what to do about it. Once I know more about that, I'll try to fix this rule